pkg/reconciler/taskrun/cancel.go

```diff
@@ -22,32 +22,26 @@ import (
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/kubernetes"
 	"knative.dev/pkg/apis"
 )
 
-type logger interface {
-	Warn(args ...interface{})
-	Warnf(template string, args ...interface{})
-}
-
 // cancelTaskRun marks the TaskRun as cancelled and delete pods linked to it.
-func cancelTaskRun(tr *v1alpha1.TaskRun, clientSet kubernetes.Interface, logger logger) error {
-	logger.Warn("task run %q has been cancelled", tr.Name)
+func (c *Reconciler) cancelTaskRun(tr *v1alpha1.TaskRun) error {
```

Comment:
:fearful:

Reply:
Resolved! This change also let me remove a gross typecast in `taskrun_test.go`, so 👍 thanks for this recommendation.